internal/transport/http2_client.go

```diff
@@ -861,6 +861,10 @@ func (t *http2Client) Close() error {
 	}
 	t.mu.Unlock()
 	t.controlBuf.finish()
```

Comment:
It's possible this exits while loopywriter is about to write to the connection, I believe.

You may want to add something to `loopyWriter.run()` that closes a channel / fires an event when it exits via a defer, and then block on that.

Reply:
Yeah, this is what I figured. Will get to this after finishing up Easwar's comments on #4387.

Reply:
Will add an event.